docs/dev/design/new-notifications-system.rst

```diff
@@ -0,0 +1,459 @@
+* In case we need a pretty specific order, we can add an extra field for that, but it adds unnecessary complexity at this point.
+* For those notifications that are attached to the ``Project`` or ``Organization``, should it be shown to all the members even if they don't have admin permissions?
+  If yes, this is good because all of them will be notified but only some of them will be able to take an action.
+  If no, non-admin users won't see the notification and won't be able to communicate this to the admins
+* Notification could be attached to a ``BuildCommand`` in case we want to display a specific message on a command itself.
+  We don't know how useful this will be, but it's something we can consider in the future.
+* Notification preferences: what kind of notifications I want to see in my own bell icon?
+
+  - Build errors
+  - Build tips
+  - Product updates
+  - Blog post news
+  - Organization updates
+  - Project updates
+
+
+Implementation ideas
+--------------------
+
+This section shows all the classes and models involved for the notification system as well as some already known use-cases.
+
+.. note:: Accessing the database from the build process
+
+    Builders doesn't have access to the database due to security reasons.
+    We had solved this limitation by creating an API endpoint the builder hits once they need to interact with the databse to get a ``Project``, ``Version`` and ``Build`` resources, create a ``BuildCommand`` resource, etc.
+
+    Besides, the build process is capable to trigger Celery tasks that are useful for managing more complex logic
+    that also require accessing from and writing to the database.
+
+    Currently, ``readthedocs.doc_builder.director.Director`` and
+    ``readthedocs.doc_builder.environments.DockerBuildEnvironment``
+    have access to the API client and can use it to create the ``Notification`` resources.
+
+    I plan to use the same pattern to create ``Notification`` resources by hitting the API from the director or the build environment.
+    In case we require hitting the API from other places, we will need to pass the API client instance to those other classes as well.
+
+
+
+``Message`` class definition
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This class encapsulates the content of the notification (e.g. header, body, icon, etc) --the message it's shown to the uer--,
+and some helper logic to return in the API response.
+
+.. code-block:: python
+
+    class Message:
+        def __init__(self):
+            header = str
+            body = str
+            icon = str
+            color = str
+            type = str(ERROR, WARINIG, NOTE, TIP)
+
+        def get_display_icon(self):
+            if self.icon:
+                return self.icon
+
+            if self.type == ERROR:
+                return "fa-exclamation"
+            if self.type == WARNING:
+                return "fa-triangle-exclamation"
```

Comment:
These should also include the icon style type too -- `solid` and `duotone`. I use duotone most frequently, but there are some icons that are more legible as `solid`, and this is done on a per-icon basis. This is especially true when the icons are used < 24px.

Reply:
Great. I added `icon_style` in 3ff602be9